stonesoup/sensor/base.py

```diff
@@ -1,43 +1,87 @@
 # -*- coding: utf-8 -*-
-from abc import abstractmethod
+import weakref
+from abc import abstractmethod, ABC
+
+from ..types.array import StateVector
+from ..platform import Platform
 
 from ..base import Base, Property
-from ..types.state import StateVector
 
 
-class Sensor(Base):
+class BaseSensor(Base, ABC):
     """Sensor base class
 
-    A sensor object that operates according to a given
-    :class:`~.MeasurementModel`.
+    .. warning::
+        This class is private and should not be used or subclassed directly. Instead use the
+        :class:`~.Sensor` class which is needed to achieve the functionality described in this
+        class's documentation.
+
     """
+    platform_system = Property(weakref.ref, default=None,
```

Comment:
As in could it be a `property`? If so then I think so. There is no particular reason this needs to be set by a user or set in the `__init__`, so yes, it could be changed.

Reply:
Actioned. I think you're right it makes more sense that way.